src/Stryker.Core/Stryker.Core/StrykerRunner.cs

```diff
@@ -100,11 +103,33 @@ public StrykerRunResult RunMutationTest(IStrykerInputs inputs, ILoggerFactory lo
                 // Test
                 foreach (var project in _mutationTestProcesses)
                 {
-                    project.Test(project.Input.ProjectInfo.ProjectContents.Mutants.Where(x => x.ResultStatus == MutantStatus.NotRun).ToList());
+                    project.Test(project.Input.SourceProjectInfo.ProjectContents.Mutants.Where(x => x.ResultStatus == MutantStatus.NotRun).ToList());
                     project.Restore();
                 }
 
-                reporters.OnAllMutantsTested(readOnlyInputComponent);
+                var combinedTestProjectsInfo = _mutationTestProcesses.Select(mtp => mtp.Input.TestProjectsInfo).Aggregate((a, b) => a + b);
+                var testRunnerPools = _mutationTestProcesses.Select(mtp => mtp.Input.TestRunner).Cast<VsTestRunnerPool>();
+
+                foreach(var testRunner in testRunnerPools)
+                {
+                    foreach(var (_, vsTestDescription) in testRunner.VsTests)
+                    {
+                        var vsTestCase = vsTestDescription.Case;
+                        var testFile = combinedTestProjectsInfo.TestFiles.SingleOrDefault(t => t.FilePath == vsTestCase.CodeFilePath);
+                        if(testFile is not null)
+                        {
+                            testFile.Tests.Add(new TestCase
+                            {
+                                Id = vsTestCase.Id,
+                                Name = vsTestCase.FullyQualifiedName,
+                                Source = null,
```

Comment:
Maybe we can create a seperate story to also fill the source with the actual test code?

Reply:
I could not yet find the test source with the information we currently have available but this story needs major rework anyway. I'm thinking of splitting this in many different improvements pr's because I think the test info is split in 30 classes right now.